polus-feature-extraction-plugin/src/utility.py

```diff
@@ -0,0 +1,1035 @@
+                df = analysis.feature_extraction()
+                
+                #Check whether csvfile is csvone to save the features extracted from all the images in same csv file
+                if csvfile == 'csvone':
+                    df_feature = df_feature.append(df)
+                else:
+                    df_feature = df
+                index+=1
+                print("Number of images processed-",index)
+
+
+        jutil.kill_vm()#kill the vm
+        return(df_feature,seg_filenames1)#return dataframe and the segment imag filename
+        
+#create class to extract features from the images      
+class Analysis(ConvertImage):
+    
+    def __init__(self,label_image,features, seg_file_names1, csvfile,outDir,boxSize,angleDegree, pixelDistance,intensity_image=None):
+        self.objneighbors = []
+        self.numneighbors = []
+        self.labels = []
+        self.uniqueindices_list=[]
+        self.meanind_list=[]
+        self.rot_position =[]
+        self.rot_list =[]
+        self.sub_rot_list=[]
+        self.feretdiam=[]
+        self.area_list=[]
+        self.perim_list=[]
+        self.df_insert = pd.DataFrame([])
+        self.df_csv = pd.DataFrame([])
+        self.boxsize = boxSize #box size to get the perimeter for calculating neighbors and feret diameter
+        self.thetastart = 1
+        self.thetastop =  angleDegree # angleDegree+1#since python indexing is from 0, to calculate for 180 degree have added 1
+        self.pixeldistance = pixelDistance
+        if self.pixeldistance is None:
+            self.pixeldistance = 5
+        else:
+            self.pixeldistance = pixelDistance
+        if self.boxsize is None:
+            self.boxsize = 3
+        else:
+            self.boxsize = boxSize
+        if self.thetastop is None:
+            self.thetastop = 180
+        else:
+            self.thetastop = angleDegree
+        self.feature = features# list of features to calculate
+        self.csv_file = csvfile#save the features(as single file for all images or 1 file for each image) in csvfile
+        self.output_dir = outDir#directory to save output
+        
+        self.label_image = label_image#assign labeled image
+        del label_image
+        
+        self.intensity_image = intensity_image#assign intensity image
+        del intensity_image
+        
+        self.filenames = seg_file_names1#assign filenames
+        
+    #function to get borders for calculating neighbors and feret diameter
+    def box_border_search(self, label_image, boxSize):
+    #Get image shape values
+        height, width = label_image.shape
+
+    #Get boxsize values
+        floor_offset = math.floor(self.boxsize/2)
+        ceil_offset = math.ceil(self.boxsize/2)
+
+    #Create the integral image
+        int_image = np.zeros((height+1, width+1))
+        int_image[1:,1:] = np.cumsum(np.cumsum(np.double(label_image),0),1)
+        int_image_transpose = int_image.T
+        del int_image
```

Comment:
A large percentage of your code is delete statements. Can you try to aggregate all delete statements to the end of each comment section? Also, wrap multiple objects into a single del statement.

For example, instead of:
```python
del int_image
int_image_int = int_image_transpose.astype(int)
del int_image_transpose
```

Do this:
```python
int_image_int = int_image_transpose.astype(int)
del int_image,int_image_transpose
```